lib/fog/azurerm/requests/storage/upload_block_blob_from_file.rb

```diff
@@ -0,0 +1,59 @@
+module Fog
+  module Storage
+    class AzureRM
+      # This class provides the actual implemention for service calls.
+      class Real
+        SINGLE_BLOB_PUT_THRESHOLD = 32 * 1024 * 1024
+        BLOCK_SIZE = 4 * 1024 * 1024
+
+        def upload_block_blob_from_file(container_name, blob_name, file_path, options = {})
+          Fog::Logger.debug "Uploading file #{file_path} as blob #{blob_name} to the container #{container_name}."
+          if file_path.nil?
+            Fog::Logger.debug "Blob #{blob_name} created successfully."
+            blob = @blob_client.create_block_blob container_name, blob_name, nil, options
+            return blob
+          end
+
+          begin
+            size = File.size file_path
+
+            if size <= SINGLE_BLOB_PUT_THRESHOLD
+              blob = @blob_client.create_block_blob container_name, blob_name, IO.binread(File.expand_path(file_path)), options
+            else
+              blocks = []
+              File.open file_path, 'rb' do |file|
+                while (read_bytes = file.read(BLOCK_SIZE))
+                  block_id = Base64.strict_encode64 random_string(32)
+                  @blob_client.put_blob_block container_name, blob_name, block_id, read_bytes, options
+                  blocks << [block_id]
```

Comment:
No, the design for the blocks is:
A ordered list of lists in the following format:
```ruby
[ ["block_id1", :committed], ["block_id2", :uncommitted], ["block_id3"], ["block_id4", :committed]... ]
```
The first element of the inner list is the block_id, the second is optional and can be either :committed or :uncommitted to indicate in which group of blocks  the id should be looked for. If it is omitted, the latest of either group will be used.